### DISCUSSION TITLE
Unable to override RabbitMQConnection in AppHost

### DISCUSSION THREAD

**Discussion by lorenzoponsbarber** [microsoft/aspire#4389] [Q&A] [answered]
Good morning,

I don't know if it's my fault or a bug, but I can't override the RabbitMQ connectionstring from AppHost: 
[https://learn.microsoft.com/en-us/dotnet/aspire/messaging/rabbitmq-client-component?tabs=dotnet-cli#use-a-connection-string](https://learn.microsoft.com/en-us/dotnet/aspire/messaging/rabbitmq-client-component?tabs=dotnet-cli#use-a-connection-string)

I have tried with these two settings in the appsettings.json of AppHost (if I use `builder.AddConnectionString("RabbitMQConnection")` it works correctly)

My appsettings.json
```
{
  "Logging": {
    "LogLevel": {
      "Default": "Information",
      "Microsoft.AspNetCore": "Warning",
      "Aspire.Hosting.Dcp": "Warning"
    }
  },
  "ConnectionStrings": {
    "RabbitMQConnection": "amqp://guest:guest@localhost:5672"
  },
  "Aspire": {
    "RabbitMQ": {
      "Client": {
        "ConnectionString": "amqp://guest:guest@localhost:5672"
      }
    }
  }
}
```

**Comment by davidfowl** [comment]
Can you share the full example? When you call AddConnectionString and you pass it to the WithReference method, it’ll make sure that value has been properly set, or there will be an error the dashboard that explains what to do.

In the apphost the ConnectionStrings section is used

**Reply by lorenzoponsbarber** [reply]
Good morning,

As I mentioned before, the error is not with `builder.AddConnectionString("RabbitMQConnection")`.

The error is when I do `builder.AddRabbitMQ("RabbitMQConnection")`, it does not get the connectionstring from the configuration.


I have attached an example: [RabbitError.zip](https://github.com/user-attachments/files/15604833/RabbitError.zip)
Thank you!

**Reply by davidfowl** [reply]
Sounds about right. When you call AddRabbitMQ it will use a container instead of using the connection string from configuration. The documentation for that method is here https://learn.microsoft.com/en-us/dotnet/api/aspire.hosting.rabbitmqbuilderextensions.addrabbitmq?view=dotnet-aspire-8.0.1.

> Adds a RabbitMQ container to the application model.

**Reply by lorenzoponsbarber** [reply]
I'm sorry, but I don't understand your answer...

**Reply by davidfowl** [reply]
1. [AddRabbitMQ](https://learn.microsoft.com/en-us/dotnet/api/aspire.hosting.rabbitmqbuilderextensions.addrabbitmq?view=dotnet-aspire-8.0.1) - Use this when you want to spin up a rabbit MQ container. Using this method will not read a connection string from configuration.

2. [AddConnectionString](https://learn.microsoft.com/en-us/dotnet/api/aspire.hosting.parameterresourcebuilderextensions.addconnectionstring?view=dotnet-aspire-8.0.1#aspire-hosting-parameterresourcebuilderextensions-addconnectionstring(aspire-hosting-idistributedapplicationbuilder-system-string-system-string)) - This method will read the connection string from the apphost's configuration.

**Reply by lorenzoponsbarber** [reply]
So, is it a documentation bug or a RabbitMQ component bug?

> When using a connection string from the ConnectionStrings configuration section, you can provide the name of the connection string when calling builder.AddRabbitMQ

[https://learn.microsoft.com/en-us/dotnet/aspire/messaging/rabbitmq-client-component?tabs=dotnet-cli#use-a-connection-string](https://learn.microsoft.com/en-us/dotnet/aspire/messaging/rabbitmq-client-component?tabs=dotnet-cli#use-a-connection-string)

**Reply by davidfowl** [reply]
Yes, that looks incorrect.

**Reply by davidfowl** [reply, marked as answer]
I filed https://github.com/dotnet/docs-aspire/issues/1054
